### DISCUSSION TITLE
Cannot move location of data files (Permission denied (13))

### DISCUSSION THREAD

**Discussion by kdsub1** [haugene/docker-transmission-openvpn#2624] [Q&A] [answered]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [ ] I have tried using dev branch

### Docker run config used

version: '3.3'
services:
    transmission-openvpn:
        cap_add:
            - NET_ADMIN
        volumes:
            - '/volume1/docker/transmission-openvpn/:/data'
            - '/volume1/torrents/:/config'
        environment:
            - OPENVPN_PROVIDER=#omitted
            - OPENVPN_CONFIG=Hong.Kong
            - OPENVPN_USERNAME=#omitted
            - OPENVPN_PASSWORD=#omitted 
            - PUID=#omitted 
            - TZ=Asia/Hong_Kong
            - TRANSMISSION_WEB_UI=flood-for-transmission
        logging:
            driver: json-file
            options:
                max-size: 10m
        ports:
            - ''#omitted
        image: haugene/transmission-openvpn

### Current Behavior

I am trying to use the "Set Location" option to move the completed data files of torrents to another folder within the "data/complete" directory (main folder where all downloads end up).

I've given read/write permission to all folders/subfolders within this directory to the User in my Synology NAS that operates Transmission (the user I put the PUID under in docker compose). This user is a non-admin/root user btw. 

I've added a "Common Path" to the destination folder I want to move the files to (for example "/completed/1.collections"). I've also used the full 'Location' string of where this folder is on my NAS. No luck.

When I select the new Torrent Location (detailed above), and click Set Torrent Location (Move Data option is checked), the torrent stops and I get an error message: Couldn't move '/data/completed' to '/completed/1.collections': Permission denied (13)

How can I move the completed torrent data files to a new folder?

Please note: 1) everything else works perfectly! I am amazed at how great this is, thank you, 2) I am so new to NAS, Docker and everything - I just bought my Synology last week! Really sorry if this is a basic question.

### Expected Behavior

Be able to move torrent data files to new sub folders within the "completed" directory.

### How have you tried to solve the problem?

1) Given read/write permission to User that is attached to Transmission/docker. Also made the folder owner this User
2) Tried different arrays of the new Path I want to move the files to.
3) Used different folders (besides the "1.collections" folder. Always says Permission denied for every folder.

### Log output

<img width="535" alt="Screenshot 2023-05-27 at 7 52 43 PM" src="https://github.com/haugene/docker-transmission-openvpn/assets/53335201/9c3cbbeb-c40b-42bb-a86f-8fe573aae6b6">

### HW/SW Environment

```markdown
- OS: Synology DS423+
- Docker: Standalone (20.10.3). Using Portainer.
```

**Comment by ilike2burnthing** [comment, marked as answer]
> I am trying to use the "Set Location" option to move the completed data files of torrents to another folder within the "data/complete" directory (main folder where all downloads end up).

> Couldn't move '/data/completed' to '/completed/1.collections': Permission denied (13)

Should the new location instead be `/data/completed/1.collections` if it's supposed to be within your `/data/completed` folder?

**Comment by kdsub1** [comment]
> > I am trying to use the "Set Location" option to move the completed data files of torrents to another folder within the "data/complete" directory (main folder where all downloads end up).
> 
> > Couldn't move '/data/completed' to '/completed/1.collections': Permission denied (13)
> 
> Should the new location instead be `/data/completed/1.collections` if it's supposed to be within your `/data/completed` folder?

Thanks for the reply! I still want to keep and have the downloads be in `/data/completed`, however, I want to manually move only a select few videos to `/data/completed/1.collections`

Does this mean that I create a new line in docker compose to include the new folder as a volume? Like below:

volumes:
- '/volume1/docker/transmission-openvpn/:/data'
- '/volume1/torrents/:/config'
- '/volume1/docker/transmission-openvpn/:/data/completed/1.collections'

Just a bit reluctant to change anything in docker compose as currently everything (else) works flawlessly. Don't want to mess that up -_-

**Comment by ilike2burnthing** [comment]
The error above shows that you're trying to move it to `/completed/1.collections` instead of `/data/completed/1.collections`.

**Reply by kdsub1** [reply]
> The error above shows that you're trying to move it to `/completed/1.collections` instead of `/data/completed/1.collections`.

Sorry don't know what I was thinking when I wrote this - was copy/pasting the wrong paths. 

Anyways, when I try to make a 'Common Path' of data/completed/1.collections I get a warning saying "Path must be an absolute path"

<img width="488" alt="Screenshot 2023-05-30 at 1 01 38 PM" src="https://github.com/haugene/docker-transmission-openvpn/assets/53335201/2fed74fe-829f-4ef8-a3b9-f85f5d9e6f2c">

**Comment by pkishino** [comment]
Moved as not issue but support question

**Comment by kdsub1** [comment]
oh my. embarrassed to say how I fixed it. But I restarted the docker stack and now it works fine 🫠
